Fix at-font-face-descriptors.html to not expect clamped specified value for font-style

nt1m opened the pull request:
#### c3e93693835f3f31a9667552bc1aaf41caf8bb2d
<pre>
Fix at-font-face-descriptors.html to not expect clamped specified value for font-style
<a href="https://bugs.webkit.org/show_bug.cgi?id=246960">https://bugs.webkit.org/show_bug.cgi?id=246960</a>
rdar://101517323

Reviewed by Darin Adler.

css/css-fonts/variations/at-font-face-descriptors.html assumes `font-style: oblique calc(91deg)` gets serialized to `font-style: oblique 90deg` in the specified style, which should not be the case according to the spec: <a href="https://drafts.csswg.org/css-values-3/#calc-range">https://drafts.csswg.org/css-values-3/#calc-range</a>

&gt; (Clamping is not performed on specified values.)

* LayoutTests/imported/w3c/web-platform-tests/css/css-fonts/variations/at-font-face-descriptors-expected.txt:
* LayoutTests/imported/w3c/web-platform-tests/css/css-fonts/variations/at-font-face-descriptors.html:

Canonical link: <a href="https://commits.webkit.org/255943@main">https://commits.webkit.org/255943@main</a>
</pre>
<!--EWS-Status-Bubble-Start-->
https://github.com/WebKit/WebKit/commit/1e3841f9edd0a7bb32f55226eaf8a60e54b7a896

| Misc | iOS, tvOS & watchOS  | macOS  | Linux |  Windows |
| ----- | ---------------------- | ------- |  ----- |  --------- |
| [✅ 🧪 style](https://ews-build.webkit.org/#/builders/6/builds/94079 "Passed style check") | [✅ 🛠 ios](https://ews-build.webkit.org/#/builders/77/builds/3269 "Built successfully") | [✅ 🛠 mac](https://ews-build.webkit.org/#/builders/43/builds/24621 "Built successfully") | [✅ 🛠 wpe](https://ews-build.webkit.org/#/builders/8/builds/103704 "Built successfully") | [✅ 🛠 🧪 win](https://ews-build.webkit.org/#/builders/10/builds/164047 "Built successfully and passed tests") 
| | [✅ 🛠 ios-sim](https://ews-build.webkit.org/#/builders/76/builds/3285 "Built successfully") | [✅ 🛠 mac-debug](https://ews-build.webkit.org/#/builders/71/builds/31493 "Built successfully") | [✅ 🛠 gtk](https://ews-build.webkit.org/#/builders/36/builds/86386 "Built successfully") | [✅ 🛠 wincairo](https://ews-build.webkit.org/#/builders/12/builds/99747 "Built successfully") 
| [✅ 🧪 webkitperl](https://ews-build.webkit.org/#/builders/19/builds/99738 "Passed tests") | [✅ 🧪 ios-wk2](https://ews-build.webkit.org/#/builders/78/builds/2356 "Passed tests") | [✅ 🛠 mac-AS-debug](https://ews-build.webkit.org/#/builders/61/builds/80500 "Built successfully") | [✅ 🧪 gtk-wk2](https://ews-build.webkit.org/#/builders/35/builds/29370 "Passed tests") | 
| | [✅ 🧪 api-ios](https://ews-build.webkit.org/#/builders/9/builds/84282 "Passed tests") | [  ~~🧪 api-mac~~](https://ews-build.webkit.org/#/builders/3/builds/83972 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [✅ 🧪 api-gtk](https://ews-build.webkit.org/#/builders/34/builds/72322 "Passed tests") | 
| | [✅ 🛠 tv](https://ews-build.webkit.org/#/builders/66/builds/37880 "Built successfully") | [  ~~🧪 mac-wk1~~](https://ews-build.webkit.org/#/builders/73/builds/17795 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | | 
| | [✅ 🛠 tv-sim](https://ews-build.webkit.org/#/builders/67/builds/35758 "Built successfully") | [✅ 🧪 mac-wk2](https://ews-build.webkit.org/#/builders/70/builds/19061 "Passed tests") | | 
| [✅ 🛠 🧪 merge](https://ews-build.webkit.org/#/builders/74/builds/4106 "Built successfully and passed tests") | [✅ 🛠 watch](https://ews-build.webkit.org/#/builders/63/builds/39633 "Built successfully") | [  ~~🧪 mac-AS-debug-wk2~~](https://ews-build.webkit.org/#/builders/60/builds/41634 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | | 
| | [✅ 🛠 watch-sim](https://ews-build.webkit.org/#/builders/65/builds/41573 "Built successfully") | [✅ 🧪 mac-wk2-stress](https://ews-build.webkit.org/#/builders/62/builds/38294 "Passed tests") | | 
<!--EWS-Status-Bubble-End-->

darinadler commented on LayoutTests/imported/w3c/web-platform-tests/css/css-fonts/variations/at-font-face-descriptors-expected.txt:
```diff
@@ -71,9 +71,9 @@ PASS font-style(valid): 'oblique' followed by zero degrees: oblique 0deg
-FAIL font-style(valid): 'oblique' followed by  angle in radians: oblique 0rad assert_equals: Unexpected resulting value. expected "oblique 0deg" but got "oblique 0rad"
+PASS font-style(valid): 'oblique' followed by calc with out of range value (should be clamped): oblique calc(91deg)
+PASS font-style(valid): 'oblique' followed by calc with out of range value (should be clamped): oblique calc(-91deg)
+FAIL font-style(valid): 'oblique' followed by angle in radians: oblique 0rad assert_equals: Unexpected resulting value. expected "oblique 0deg" but got "oblique 0rad"
```
I wonder if this is correct!?

nt1m replied:
@darinadler This seems correct according to the spec, the spec says the canonical unit for `<angle>` is `deg`, which means it should always be used for serialization.

https://w3c.github.io/csswg-drafts/css-values/#angles

> All [<angle>](https://w3c.github.io/csswg-drafts/css-values/#angle-value) units are [compatible](https://w3c.github.io/csswg-drafts/css-values/#compatible-units), and [deg](https://w3c.github.io/csswg-drafts/css-values/#deg) is their [canonical unit](https://w3c.github.io/csswg-drafts/css-values/#canonical-unit).

darinadler replied:
Sure, but the canonical unit affects **computed** values. This test is examining **specified** values by reading back a style property, not computed values by calling getComputedStyle.

nt1m replied:
Oh I missed that, we probably want to change the test then. There is some discussion around how we should consider `@font-face` descriptors in https://github.com/w3c/csswg-drafts/issues/7964 , since they're not clearly "computed" or "specified", but I lean towards "specified" given the way they're being set/read in this test.

nt1m committed:
Fix at-font-face-descriptors.html to not expect clamped specified value for font-style
https://bugs.webkit.org/show_bug.cgi?id=246960
rdar://101517323

Reviewed by Darin Adler.

css/css-fonts/variations/at-font-face-descriptors.html assumes `font-style: oblique calc(91deg)` gets serialized to `font-style: oblique 90deg` in the specified style, which should not be the case according to the spec: https://drafts.csswg.org/css-values-3/#calc-range

> (Clamping is not performed on specified values.)

* LayoutTests/imported/w3c/web-platform-tests/css/css-fonts/variations/at-font-face-descriptors-expected.txt:
* LayoutTests/imported/w3c/web-platform-tests/css/css-fonts/variations/at-font-face-descriptors.html:

Canonical link: https://commits.webkit.org/255943@main
